pandas/_typing.py

```diff
@@ -96,3 +96,11 @@
 # DataFrame::sort_index, among others
 ValueKeyFunc = Optional[Callable[["Series"], Union["Series", AnyArrayLike]]]
 IndexKeyFunc = Optional[Callable[["Index"], Union["Index", AnyArrayLike]]]
+
+# types of `func` kwarg for DataFrame.aggregate and Series.aggregate
+AggFuncTypeBase = Union[Callable, str]
```

Comment:
If there is a more generic name you can think of than `AggFuncTypeBase` I think would be useful in other areas that aren't aggregations; i.e. in groupby a lot we accept a callable / str and resolve the latter to a builtin or NumPy func if we can, which could use this same type